graphblas/core/matrix.py

```diff
@@ -2805,6 +2805,124 @@ def power(self, n, op=semiring.plus_times):
             dtype=self.dtype,
         )
 
+    def setdiag(self, values, k=0, *, mask=None, accum=None, clear_missing=True, **opts):
+        """Set k'th diagonal with a Scalar, Vector, or array.
+
+        This is not a built-in GraphBLAS operation. It is implemented as a recipe.
+
+        Parameters
+        ----------
+        values : Vector or list or np.ndarray or scalar
+            New values to assign to the diagonal. The length of Vector and array
+            values must match the size of the diagonal being assigned to.
+        k : int, default=0
+            Which diagonal or off-diagonal to set. For example, set the elements
+            ``A[i, i+k] = values[i]``. The default, k=0, is the main diagonal.
+        mask : Mask, optional
+            Vector or Matrix Mask to control which diagonal elements to set.
+            If it is Matrix Mask, then only the diagonal is used as the mask.
```

Comment:
I like this approach for the mask. We should look into a similar approach for assignment with a row or column if we don't already do this.

Reply:
Yup, I borrowed this behavior from row/col assign, which already accept matrix masks :)